drivers/digital-io/max149x6/README.rst

```diff
@@ -0,0 +1,322 @@
+			},
+			{
+				.enabled = true,
+				.funtion = MAX14906_IN,
+			},
+		},
+	};
+
+	struct iio_app_desc *app;
+	struct iio_app_init_param app_init_param = { 0 };
+
+	ret = max14906_iio_init(&max14906_iio_desc, &max14906_iio_ip);
+	if (ret)
+		goto error;
+	
+	struct iio_app_device iio_devices[] = {
+		{
+			.name = "max14906",
+			.dev = max14906_iio_desc,
+			.dev_descriptor = max14906_iio_desc->iio_dev,
+		}
+	};
+
+	app_init_param.devices = iio_devices;
+	app_init_param.nb_devices = NO_OS_ARRAY_SIZE(iio_devices);
+	app_init_param.uart_init_params = max14906_uart_ip;
+
+	ret = iio_app_init(&app, app_init_param);
+	if (ret)
+		goto error;
+	
+	return iio_app_run(app);
+
+MAX14916
+--------
+* Industrial Digital Outputs
+* PLC systems
+
+MAX14916 Device Configuration
+-----------------------------
+
+Driver Initialization
+---------------------
+
+In order to be able to use the device, you will have to provide the support
+for the communication protocol (SPI).
+
+The first API to be called is **max14916_init**. Make sure that it returns 0,
+which means that the driver was initialized correctly.
+
+Channel Configuration
+---------------------
+
+Each channel can be configured as a digital output only.
+The state of a channel(voltage state) cand be read with the **max14916_ch_get**
+API and also set with **max14916_ch_set**.
+
+LED Configuration
+---------------------------
+
+Status LEDs cand be configured with **max14916_sled_set** which takes as
+parameters the device descriptor, the number of the channel requested for
+the LED state to change and the requested state.
+
+Fault LEDs state however can not be set by the user since they indicate
+the case of a fault, still they have a configurable minimum on-time value
+assigned to them which the user can change through the
+**max14916_fled_time_set** API.
+
+Interrupts Configuration
+------------------------
+
+The device allows the usage of interrupts which can be mapped to FAULT pin
+and is configured like any other no-os interrupt(using no_os_irq API) since
+the descriptor is common for both MAX14916 and MAX14906.
+
+MAX14916 Driver Initialization Example
+--------------------------------------
+
+.. code-block:: bash
+
+	struct max149x6_desc *max14916;
+	struct no_os_spi_init_param spi_ip = {
+		.device_id = 1,
+		.extra = &max14916_spi_extra,
+		.max_speed_hz = 100000,
+		.platform_ops = &max_spi_ops,
+		.chip_select = 1,
+	};
+	struct max149x6_init_param max14916_ip = {
+		.chip_address = 0;
+		.comm_desc = &spi_ip;
+		.crc_en = true;
+	}
+
+	ret = max14916_init(&max14916, &max14916_ip);
+	if (ret)
+		goto error;
+	
+MAX14916 no-OS IIO support
+--------------------------
+
+The MAX14916 IIO driver comes on top of the MAX14916 driver and offers support
+for interfacing IIO clients through IIO lib.
+
+MAX14916 IIO Devce Configuration
+--------------------------------
+
+Device Attributes
+-----------------
+
+MAX14916 has only 3 attributes :
+
+raw - indicates the state of a channel(0 or 1).
```

Comment:
these 3 don't render as you would like, they should probably be listed with bullets